cylc/flow/data_store_mgr.py

```diff
@@ -2428,11 +2428,25 @@ def delta_task_output(
         tp_delta = self.updated[TASK_PROXIES].setdefault(
             tp_id, PbTaskProxy(id=tp_id))
         tp_delta.stamp = f'{tp_id}@{update_time}'
-        output = tp_delta.outputs[label]
-        output.label = label
-        output.message = message
-        output.satisfied = outputs.is_message_complete(message)
-        output.time = update_time
+
+        for _label in tproxy.outputs:
+            output = tp_delta.outputs[_label]
+
+            # set the new output
+            if _label == label:
+                output.label = label
+                output.message = message
+                output.satisfied = outputs.is_message_complete(message)
+
+            # ensure all outputs are included in the delta
+            else:
+                _output = tproxy.outputs[_label]
+                output.label = _output.label
+                output.message = _output.message
+                output.satisfied = _output.satisfied
+
+            output.time = update_time
+
         self.updates_pending = True
 
     def delta_task_outputs(self, itask: TaskProxy) -> None:
```

Comment:
Might as well only use this call...

Reply:
Maybe just call `delta_task_outputs` from within `delta_task_output`

Reply:
This is a bodge workaround, I'm not sure whether we should merge it.

The incremental update is the right thing to do (so I don't want to blast that code aside), we just didn't develop a mechanism for syncing these updates over GraphQL so the UI can't handle it.